src/feh_io/feh_reader.py

```diff
@@ -0,0 +1,106 @@
+        self.check_file_paths()
+        self.check_file_type()
+
+    # Check if file paths to headers and data exist
+    def check_file_paths(self):
+        if not os.path.exists(self.data_file):
+            raise FileNotFoundError(f"File path for data is not correctly specified: {self.data_file}")
+        elif not os.path.exists(self.header_file):
+            raise FileNotFoundError(f"File path for header is not correctly specified: {self.data_file}")
+    
+    # Check if file type is correctly specified
+    def check_file_type(self):
+        if self.file_type not in ['person', 'family']:
+            raise ValueError(f"Invalid file type: {self.file_type}." 
+                             f"Must be input as '{'person'}' or '{'family'}'.")
+
+    # Reset the data read in and the bytes read in
+    def reset_data(self):
+        self.file = None
+        self.bytes_read = 0
+
+    # Set chunk size, clear data read and reset cursor
+    def set_chunk_size(self, chunk_size: int):
+        self.chunk_size = chunk_size
+        self.reset_data()
+        
+    # Set the var_list, clear data read and reset cursor
+    def set_vars(self, var_list):
+        if not isinstance(var_list, list):
+            raise TypeError("Variables must be input as a list.")
+        self.var_list = var_list
+        self.reset_data()
+
+    # calculate how many bytes are read in for a chunk
+    def calc_bytes_read(self):
+        # Initialize family and person dictionaries and numeric year
+        year, famrec, perrec = read_header_file(self.header_file)
+        
+        if self.file_type == 'person':
+            rectype = make_rec_dtype(perrec)
+
+        elif self.file_type == 'family':
+            rectype = make_rec_dtype(famrec)
+
+        # Size of chunk = -1 means read in the entire file
+        if self.chunk_size == -1:
+            bytes_read = self.file_size
+        else:
+            bytes_read = rectype.itemsize * self.chunk_size
+
+        # bytes_read being larger than filesize means that np.fromfile 
+        # attempted to read that many bytes, but stopped early because it reached the end of the file.
+        if bytes_read + self.bytes_read > self.file_size:
+            bytes_read = self.file_size - self.bytes_read
+        return bytes_read
+    
+    # Read data file
+    def read_chunk(self):
+
+        # Read in data file, get bytes read in
+        self.file = read_feh_data_file( header_file = self.header_file, 
+                                        data_file = self.data_file,
+                                        vars = self.var_list,
+                                        file_type = self.file_type, 
+                                        count = self.chunk_size,
+                                        offset = self.bytes_read)
+
+        # Calculate size of chunk in bytes and update
+        self.bytes_read += self.calc_bytes_read()
```

Comment:
It might be better to use the actual number of bytes read rather than calculate it based on chunk size and header size. Function read_feh_data_file returns a Numpy array, which has `nbytes` attribute that indicates the size of the array in bytes.

This also makes me think about completing a read: How does the caller know that it reached the end of the file and that they have all the chunks?

Reply:
Good point! I can think of two ways to address this. One is by doing variable subsetting in `FehReader.read_chunk`, which in that case would call `read_feh_data_file` with `vars` set to `None`. The other is by `read_feh_data_file` return both the data it currently returns and the number of bytes read from the file.

I think that I prefer the former. The latter changes the signature of the function and it feels somewhat odd to return one piece of data and the size of a different piece of data.

If we decide to go with the former, we could create function `select_vars(data, vars)` in `read_feh.py`, and then call this function from both `read_chunk` and `read_feh_data_file`.

Reply:
I think the former is better as well! When I have a moment this week I will implement that and let Evy know. Shouldn't change anything about how she accesses this stuff.

Reply:
Addressed and pushed!